go/vt/vtgate/planbuilder/route_planning.go

```diff
@@ -46,7 +46,11 @@ func gen4Planner(_ string) func(sqlparser.Statement, *sqlparser.ReservedVars, Co
 }
 
 func newBuildSelectPlan(sel *sqlparser.Select, vschema ContextVSchema) (engine.Primitive, error) {
-	semTable, err := semantics.Analyse(sel, vschema)
+	keyspace, err := vschema.DefaultKeyspace()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
this will start failing for cases when there is no default database selected.

Reply:
hmmm... how come no tests are failing? do we need to add more tests to the planner?

you are probably absolutely correct about this comment. that being said - since this is not failing any tests, how about we take care of this in a separate PR? We should add more tests, and then also fix the issue

WDYT?

Reply:
I think we are introducing this check here, so it be be better to have a test that covers the error case if existing tests do not fail.